### DISCUSSION TITLE
"Display tabs and spaces" switched off and yet still showing up.

### DISCUSSION THREAD

**Discussion by scriptor-pro** [vkbo/novelWriter#1984] [Help]
![tab-space-noverwriter-2point5](https://github.com/user-attachments/assets/5ed641f1-13d8-4a27-8c24-5fc2a8644b3b)

 A tiny typographic detail in the editor/document is driving me nuts. 

I've updated/upgraded to novelwriter 2.5 10th July 2024 version. Not sure this was happening with the previous version I was using.

In Tools>Preferences (Ctrl ,) Display tabs and spaces is switched off. And yet, tabs and spaces are displayed. I find it harder to write with spaces and tabs showing up. 

If it heps : I'm using NovelWriter 2.5 on MXLinux 21 (Debian 11).

What should I do ?

**Comment by vkbo** [comment]
It's not showing "tabs and spaces". The coloured parts you're seeing are non-breaking spaces. They are highlighted. Your text seems to have a lot of those for some reason?

**Reply by vkbo** [reply]
What it looks like is that you have three spaces between each word. Non-breaking space + space + non-breaking space.

**Reply by vkbo** [reply]
I think I know what's going on. Do you have this setting enabled?

![image](https://github.com/user-attachments/assets/2c3a22a1-e5d3-48b0-96ba-636a85523a9f)

You probably have a space in the boxes, so the editor inserts non-breaking spaces around your regular spaces.

This setting should probably not allow spaces in the values. I'll look into adding a check for that.

**Reply by vkbo** [reply]
I've made a fix that makes sure the content of those two boxes consists of unique characters with no spaces. #1986. It will be included in release 2.5.1.

In the mean time, make sure there are no spaces in those settings, and your issue should go away – assuming this is really the cause, which I think it is. I managed to reproduce the issue. Note that non-breaking spaces will still be highlighted in the text, but there shouldn't be as many of them as in your screenshot.
